core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -926,16 +926,27 @@ private boolean deleteLogSegmentsDueToLeaderEpochCacheTruncation(EpochEntry earl
                 return isSegmentDeleted;
             }
 
-            private boolean deleteRemoteLogSegment(RemoteLogSegmentMetadata segmentMetadata, Predicate<RemoteLogSegmentMetadata> predicate)
+            private boolean deleteRemoteLogSegment(String topic, RemoteLogSegmentMetadata segmentMetadata, Predicate<RemoteLogSegmentMetadata> predicate)
```

Comment:
Can we take the `topic` name from the segmentMetadata instead of propagating it from the caller?

```
String topic = segmentMetadata.topicIdPartition().topic();
```